spring-shell-core/src/main/java/org/springframework/shell/Shell.java

```diff
@@ -185,10 +190,17 @@ private boolean noInput(Input input) {
 			// Try to complete arguments
 			MethodTarget methodTarget = methodTargets.get(best);
 			Method method = methodTarget.getMethod();
-			Arrays.stream(method.getParameters())
-				.map(Utils::createMethodParameter)
-				.flatMap(mp -> findResolver(mp).complete(mp, argsContext).stream())
-				.forEach(candidates::add);
+			
+			List<MethodParameter> parameters = Utils.createMethodParameters(method).collect(Collectors.toList());
+			validateParameters(parameters);
```

Comment:
I think we can move this validation at startup time when we discover MethodTargets.
Will do on merge